// expire/index.js
// MODULES //

import React from 'react';
import ReactDOM from 'react-dom';


// EXPIRE //

class Expire extends React.Component {
	/**
	* Constructor function
	*/
	constructor() {
		super();

		// Set initial component states...
		this.state = {
			visible: true
		};
	}

	/**
	* Invoked when component is receiving new children, this method sets
	*  up the time and sets the component visible before rendering.
	*
	* @param {Object} newProps - new properties
	*/
	componentWillReceiveProps( newProps ) {
		if ( newProps.children !== this.props.children ) {
			this.setTimer();
			this.setState( { visible: true } );
		}
	}

	/*
	* Invoked once on client after the initial rendering. Sets up component timer.
	*/
	componentDidMount() {
		this.setTimer();
	}

	setTimer() {
		this._timer != null ? clearTimeout( this._timer ) : null;

		this._timer = setTimeout( () => {
			this.setState( { visible: false } );
			this._timer = null;
		}, this.props.delay );
	}

	/*
	* Invoked immediately before component is unmounted from DOM. Performs clean-up of
	* component timer.
	*/
	componentWillUnmount() {
		clearTimeout( this._timer );
	}

	/**
	* React component render method
	*/
	render() {
		return this.state.visible
			? <div>{this.props.children}</div>
			: <span />;
	}
}


// DEFAULT PROPERTIES //

Expire.defaultProps = {
	delay: 1000
};


// EXPORTS //

export default Expire;
