// text-area/index.js
// MODULES //

import Radium from 'radium';
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styles from 'styles';


// TEXT AREA //

class TextArea extends Component {
	/**
	* Create a text area
	*
	* @param {Object} props - component properties (`onChange` callback and `text`)
	*/
	constructor( props ) {
		super( props );

		// Initialize state variables...
		this.state = {
			text: props.text || ''
		}

		/*
		* Event handler invoked when text area value changes. Updates `text` and invokes
		* `onChange` callback with the new text as its first argument
		*/
		this.handleChange = ( event ) => {
			let text = event.target.value
			this.setState( { 'text': text } );
			this.props.onChange( text ) ;
		}

	}

	/*
	* React component render method.
	*/
	render() {
		return (
			<textarea style={styles.feedbackTextArea} value={this.state.text} onChange={this.handleChange}>
			</textarea>
		);
	}
}


// DEFAULT PROPERTIES //

TextArea.defaultProps = {
	onChange: function(){}
}


// EXPORTS //

export default TextArea;
