// feedback-form/index.js
// MODULES //

import Radium from 'radium';
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styles from 'styles';

import TextArea from 'text-area';
import CheckBox from 'check-box';


// FEEDBACK FORM //

class FeedbackForm extends Component {
	/*
	* Create a feedback form to collect detailed user feedback.
	*/
	constructor( props ) {
		super( props );

		// Set initial states for feedback form...
		this.state = {
			comment: '',
			checked: [
				false,
				false,
				false,
				false
			]
		};

		/**
		* Event handler invoked when the value of the text area changes.
		*
		* @param {String} text - current content of the text area
		*
		* @returns {Void} updates the `comment` state of the component
		*/
		this.handleTextAreaChange = ( text ) => {
			this.setState( { 'comment': text } );
		}

		this.handleCheckBoxChange = ( id, checked ) => {
			let arr = this.state.checked.slice();
			arr[ id ] = checked;
			this.setState({
				checked: arr
			});
		}
	}

	/*
	* React component render method.
	*/
	render() {
		return (
			<div className="feedbackForm" style={[styles.feedbackForm,{ top: this.props.top }]}>
			<button ref="closeButton" style={styles.closeButton} onClick={this.props.closeHandler}>Close</button>
				<div style={styles.feedbackTitle}>Feedback</div>
				<form style={styles.feedbackFormElem}>
					<CheckBox id="0" text="I do not understand this at all." onChange={this.handleCheckBoxChange}/>
					<CheckBox id="1" text="This needs a more detailed explanation." onChange={this.handleCheckBoxChange}/>
					<CheckBox id="2" text="I can't follow the logic." onChange={this.handleCheckBoxChange}/>
					<CheckBox id="3" text="I have the following comments:" onChange={this.handleCheckBoxChange}/>
					<TextArea text={this.state.comment} onChange={this.handleTextAreaChange}></TextArea>
					<input ref="submitButton" type="submit" style={styles.submitButton} onClick={this.props.submitHandler} value="Submit" />
				</form>
			</div>
		);
	}
}


// DEFAULT PROPERTIES //

FeedbackForm.defaultProps = {
	closeHandler: function(){},
	submitHandler: function(){},
	top: 0
}


// EXPORTS //

export default Radium( FeedbackForm );
